channeldb/db.go

```diff
@@ -277,17 +345,25 @@ func CreateWithBackend(backend kvdb.Backend, modifiers ...OptionModifier) (*DB,
 		modifier(&opts)
 	}
 
+	// Add the graph's buckets to the channeldb cache.
+	cache := kvdb.NewCache(backend)
```

Comment:
Should be conditional on the DB type (remote vs local, etc).

Reply:
I agree, this probably doesn't help that much in a `bbolt` environment but does consume a lot of memory (if you factor in that `bbolt` itself does the `mmap` thing and then we add the cache on top). In a remote DB scenario we don't have the `mmap` allocation so we should already consume quite a bit less RAM which we can then invest into a cache like this.

Reply:
Does the cache with the buckets that are currently selected for caching consume a lot of memory?

Reply:
> Does the cache with the buckets that are currently selected for caching consume a lot of memory?

This is something we'll need to evaluate in the wild for larger nodes. IIUC the latest version skips problematic buckets like the revocation log (which stores a state for each channel update ever along w/ all the HTLCs).

Reply:
Chace is now restricted to remote backends.